group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -452,21 +453,39 @@ public Group group(String groupId, long committedOffset) throws GroupIdNotFoundE
     /**
      * Get the Group List.
      *
-     * @param statesFilter The states of the groups we want to list.
-     *                     If empty all groups are returned with their state.
-     * @param committedOffset A specified committed offset corresponding to this shard
+     * @param statesFilter      The states of the groups we want to list.
+     *                          If empty, all groups are returned with their state.
+     * @param typesFilter       The types of the groups we want to list.
+     *                          If empty, all groups are returned with their type.
+     * @param committedOffset   A specified committed offset corresponding to this shard.
      *
      * @return A list containing the ListGroupsResponseData.ListedGroup
      */
+    public List<ListGroupsResponseData.ListedGroup> listGroups(
+        Set<String> statesFilter,
+        Set<String> typesFilter,
+        long committedOffset
+    ) {
+        // Convert typesFilter to lowercase to make the filter case-insensitive.
+        Set<String> lowerCaseTypesFilter = typesFilter.stream()
+            .map(String::toLowerCase)
+            .collect(Collectors.toCollection(HashSet::new));
+
+        Predicate<Group> combinedFilter = group -> {
+            boolean stateCheck = statesFilter.isEmpty() || statesFilter.contains(group.stateAsString(committedOffset));
+            boolean typeCheck = lowerCaseTypesFilter.isEmpty() ||
+                lowerCaseTypesFilter.contains(group.type().toString().toLowerCase());
```

Comment:
nit: I wonder if it is worth storing the lower case type in the GroupType. What do you think? An alternative would be to convert the String to GroupType at L470 and compare enums here.

Reply:
I also considered just adding the lower case group types to the enum itself but I haven't seen that before and was also confused if it would make sense.

Reply:
For the second option my understanding is that to convert the string to groupType we would still need to convert it to lower case to compare and see if it is a valid group type. Lmk if I'm missing something more obvious

Reply:
I was also under the impression that as long as the word spells the correct type we don't care what the cases of the letters are? So currently if someone were to type "cLaSsic" that would still be valid. If we want to strictly allow only "Classic" and "classic" we would need to probably add both for each type to the enum itself. Lmk what you think?

Reply:
Sorry, I was not clear. I do think that we need to lowercase somewhere. However, I was trying to avoid doing it here while iterating over all the groups. This is a waste. This is why I suggested to store the lowercase version of the type in the enum or convert to the enum before. Do you see what I mean?

Reply:
I think I understand what you're saying. I'll add a way to parse any string and convert it to the necessary case to compare and determine which type it is in the group type enum. However, since some of these changes are on the client side can we do it in the second PR? That way I can test it as well. I could remove the lower case conversion here and then directly handle it there or we could change it there directly as an optimization just to be safe. What do you think?

Reply:
None of the required changes are on the client side or I don't see them. I would prefer to address it in this PR.

Reply:
Discussed offline. We realized we were talking about two different GroupType files. Made the changes to the server side GroupType and added string to enum conversion and comparison